fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -483,10 +489,62 @@ abstract class BaseBrowserFragment :
-                menuButton = menuButton,
-                browsingModeManager = activity.browsingModeManager,
-            )
+            ).apply {
+                composeView.setContent {
+                    FirefoxTheme {
+                        Column {
+                            if (isToolbarAtBottom) {
+                                AndroidView(factory = { _ -> browserToolbar })
+                            } else {
+                                Divider()
+                            }
+
+                            OpenTabNavBar(
+                                onBackButtonClick = {
+                                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                                        ToolbarMenu.Item.Back(viewHistory = false),
+                                    )
+                                },
+                                onBackButtonLongPress = {
+                                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                                        ToolbarMenu.Item.Back(viewHistory = true),
+                                    )
+                                },
+                                onForwardButtonClick = {
+                                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                                        ToolbarMenu.Item.Forward(viewHistory = false),
+                                    )
+                                },
+                                onForwardButtonLongPress = {
+                                    browserToolbarInteractor.onBrowserToolbarMenuItemTapped(
+                                        ToolbarMenu.Item.Forward(viewHistory = true),
+                                    )
+                                },
+                                onHomeButtonClick = {
+                                    findNavController().navigate(
+                                        BrowserFragmentDirections.actionGlobalHome(),
+                                    )
+                                },
+                                menuButton = menuButton,
+                                onTabsButtonClick = {
+                                    thumbnailsFeature.get()?.requestScreenshot()
+                                    findNavController().nav(
+                                        R.id.browserFragment,
+                                        BrowserFragmentDirections.actionGlobalTabsTrayFragment(
+                                            page = when (activity.browsingModeManager.mode) {
+                                                BrowsingMode.Normal -> Page.NormalTabs
+                                                BrowsingMode.Private -> Page.PrivateTabs
+                                            },
+                                        ),
+                                    )
+                                },
```

Comment:
to help with reviewing, could you post a link where the pre-existing click logic lives for these lambdas? That'll make it easier to make sure we're not missing anything before this moves forward.